timesketch/api/v1/resources/event.py

```diff
@@ -327,6 +327,201 @@ def get(self, sketch_id):
+
+    EVENT_FIELDS = ["_id", "_type", "_index", "attributes"]
+    ATTRIBUTE_FIELDS = ["attr_name", "attr_value"]
+    INVALID_ATTRIBUTES_NAMES = [
+        'datetime', 'timestamp', 'message', 'timestamp_desc']
+
+    MAX_EVENTS = 100000
+    MAX_ATTRIBUTES = 10
+    EVENT_CHUNK_SIZE = 1000
+
+    def _parse_request(self, flask_request):
+        """Validate and parse a POST request to add attributes.
+
+        Args:
+            flask_request (flask.Request): The unmodified request object.
+
+        Returns:
+            A dict with searchindex IDs as keys and the associated events as
+                values.
+        """
+        if not flask_request.is_json:
+            abort(
+                HTTP_STATUS_CODE_BAD_REQUEST,
+                "Request must be in JSON format."
+            )
+        form = flask_request.json
+        events = form.get("events")
+        if not events:
+            abort(
+                HTTP_STATUS_CODE_BAD_REQUEST,
+                "Request must contain an events field."
+            )
+        if not isinstance(events, list):
+            abort(
+                HTTP_STATUS_CODE_BAD_REQUEST,
+                "Events field must be a list."
+            )
+        if len(events) > self.MAX_EVENTS:
+            abort(
+                HTTP_STATUS_CODE_BAD_REQUEST,
+                f"Request exceeds maximum events to process {self.MAX_EVENTS}"
+            )
+
+        events_by_index = {}
+        for event in events:
+            for field in self.EVENT_FIELDS:
+                if field not in event:
+                    abort(
+                        HTTP_STATUS_CODE_BAD_REQUEST,
+                        f"Event missing field {field}."
+                    )
+
+            attributes = event.get("attributes")
+            if not isinstance(attributes, list):
+                abort(
+                    HTTP_STATUS_CODE_BAD_REQUEST,
+                    "Attributes must be a list."
+                )
+            if len(attributes) > self.MAX_ATTRIBUTES:
+                abort(
+                    HTTP_STATUS_CODE_BAD_REQUEST,
+                    f"Attributes for event exceeds maximum "
+                    f"{self.MAX_ATTRIBUTES}"
+                )
+            for attribute in attributes:
+                for field in self.ATTRIBUTE_FIELDS:
+                    if field not in attribute:
+                        abort(
+                            HTTP_STATUS_CODE_BAD_REQUEST,
+                            f"Attribute missing field {field}."
+                      )
+
+            if event["_index"] not in events_by_index:
+                events_by_index[event["_index"]] = []
+            events_by_index[event["_index"]].append(event)
+
+        return events_by_index
+
+    @login_required
+    def post(self, sketch_id):
+        """Handles POST requests to the resource.
+
+        Args:
+            sketch_id: Integer primary key for a sketch database model
```

Comment:
`Integer primary key for a sketch database model.`

Reply:
Done